paasta_tools/setup_prometheus_adapter_config.py

```diff
@@ -696,12 +702,15 @@ def create_prometheus_adapter_config(
             cluster=paasta_cluster,
             instance_type_class=KubernetesDeploymentConfig,
         ):
+            if not is_dry_run:
+                ensure_namespace(kube_client, namespace=instance_config.get_namespace())
```

Comment:
out of curiosity: how are we planning on generally creating these namespaces?

i wonder if we should just assume that these namespaces will exist and let this script not worry about them - especially since this script doesn't actually require the instance namespace to exist

Reply:
I thought of wherever we are calling ``ensure_namespace`` just make sure that we are calling it for all namespaces needed instead of just "paasta" namespace. So currently, we call ``ensure_namespace`` function in these files:
- setup_kubernetes_job : we took care of it in skj code by calling it for all service_instances
- setup_prometheus_adapter_config: this PR takes care of this
- setup_kubernetes_cr: we don't hardcode the namespaces here
- delete_kubernetes_deployments.py: we don't call this file anywhere
- setup_istio_mesh: I think this file is just used for testing in kubestage
I thought of this with the assumption that we already call ``ensure_namespace`` everywhere we need to have "paasta" namespace created. 
This way the service developer would need to just specify the namespace in yelp-soaconfigs and each script in paasta_tools would take care of creating the namespace if it doesn't exist. Not sure if this is a good idea or if there is a better way of approaching this

Reply:
i think the `ensure_namespace(..., "paasta")` was a bad copy-paste on my part when i was first writing this script 😆 (we should really only need to ensure that the `custom-metrics` namespace exists for this script to work)

if the new per-service namespaces are generally going to be created elsewhere, then i think it's fine to not attempt to create them here (since there should be no impact if we generate an autoscaling rule for something that doesn't exist yet)

Reply:
The new per-service namespaces should be created by skj since we call ``ensure_namespace`` there and since it runs every 30s then it should pick up fast if a namespace doesn't exist and create it. There could be a small gap in time between running skj and setup_prometheus_adapter file but I guess this gap is fine